### PR TITLE
Split USM alignment tests into small and large alignments

### PR DESCRIPTION
**Pull request by lyu**
Related-To: NEO-8431

### CODE REVIEW

**Review by yoon0805 — APPROVED** [review]
Fairly straightforward and good enough for a temporary solution for me.

**Review by nrspruit — DISMISSED** [review]

File: utils/test_harness/include/test_harness/test_harness_memory.hpp [hunk 1]
```diff
@@ -29,6 +28,11 @@ const auto memory_allocation_sizes =
 const auto memory_allocation_alignments = ::testing::Values(
```
Comment (yoon0805) [inline utils/test_harness/include/test_harness/test_harness_memory.hpp:28]: Considering that we separate small and large now, should be remove this regular alignment values here?

Comment (lyu) [reply utils/test_harness/include/test_harness/test_harness_memory.hpp:28]: These are used for a few other CTS as well

**Review by yoon0805 — APPROVED** [review]

File: utils/test_harness/include/test_harness/test_harness_memory.hpp (hunk 1 above)
Comment (yoon0805) [reply utils/test_harness/include/test_harness/test_harness_memory.hpp:28]: Thanks, that's good enough, so please Resolve this conversation to proceed further.

**Review by nrspruit — DISMISSED** [review]

**Review by nrspruit — APPROVED** [review]
LGTM

**Pull request merged**